controllers/bootstrap/init.go

```diff
@@ -1158,11 +1189,113 @@ func (b *Bootstrap) DeployCertManagerCR() error {
 	return nil
 }
 
+// CleanNamespaceScopeResources will delete the v3 NamesapceScopes resources and namespace scope operator
+// NamespaceScope resources include common-service, nss-managedby-odlm, nss-odlm-scope, and odlm-scope-managedby-odlm
+func (b *Bootstrap) CleanNamespaceScopeResources() error {
+	// Get the common-service OperandRegistry
+	operandRegistry, err := b.GetOperandRegistry(ctx, constant.MasterCR, b.CSData.ServicesNs)
+	if err != nil && operandRegistry == nil {
+		klog.Errorf("Failed to get common-service OperandRegistry: %v", err)
+		return err
+	}
+
+	// List all requested operators
+	if operandRegistry.Status.OperatorsStatus != nil {
+		for operator := range operandRegistry.Status.OperatorsStatus {
+			// If there is a requested operator's installMode is "no-op", then skip call delete function
+			for _, op := range operandRegistry.Spec.Operators {
+				if op.Name == operator && op.InstallMode == "no-op" {
+					klog.Infof("The v3 operator %s is still requested, skip cleaning the resources", operator)
+					return nil
+				}
+			}
+		}
+	}
+
+	// If the topology is Simple or All Namespaces Mode, delete all NamespaceScope resources
+	if b.CSData.WatchNamespaces == "" || len(strings.Split(b.CSData.WatchNamespaces, ",")) == 1 {
+
```

Comment:
I am wondering if we could has this condition chcked at the beginning of the function.
```golang
func (b *Bootstrap) CleanNamespaceScopeResources() error {

   // If the topology is (NOT ALL NS Mode) and (NOT Simple) , return
   if b.CSData.WatchNamespaces != "" && len(strings.Split(b.CSData.WatchNamespaces, ",")) > 1 {
      return nil
   }


   // Get the common-service OperandRegistry
  operandRegistry, err := b.GetOperandRegistry(ctx, constant.MasterCR, b.CSData.ServicesNs)
  if err != nil && operandRegistry == nil {
  klog.Errorf("Failed to get common-service OperandRegistry: %v", err)
  return err
  }
  ...
}
```